adagrams/game.py

```diff
@@ -1,13 +1,152 @@
 from random import randint
+    # from tests.test_wave_01 import LETTER_POOL
+
+LETTER_POOL = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+}
 
 def draw_letters():
-    pass
+    letter_pool_list = []
+    for letter, count in LETTER_POOL.items():
+        letter_pool_list.extend([letter] * count)
+        # transfer the dictionary to a 98 letters list to use random.randint
+
+    draw_letter_pool = letter_pool_list.copy()
+    # make a copy list for poping letter to avoid draw a letter and hands repeatly
+    # use index / .pop() use index
+
+    draw_hand = []
+
+    while len(draw_hand) < 10 :
+        random_index = randint(0 ,len(letter_pool_list.copy())-1)
+        draw_letter = letter_pool_list.copy()[random_index]
+        draw_hand.append(letter_pool_list.pop(random_index))
+
+    return draw_hand
+
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # letters = ["D", "O", "G", "X", "X", "X", "X", "X", "X", "X"]
+    #     word = "DOG"
+    # convert it to dictionaries to mapping the word and letters
+    uppercase_word = word.upper()
+    check_word = {}
+
+    for letter in uppercase_word:
+        if letter in check_word:
+            check_word[letter] += 1
+        else:
+            check_word[letter] = 1
+
+    letter_freq = {}
+
+    for letter in letter_bank:
+        if letter in letter_freq:
+            letter_freq[letter] += 1
+        else:
+            letter_freq[letter] = 1
+
+    # mapping1 : to see word(check_word) has the keys of letter_bank(letter_freq) or not, Not = False
+    # mapping2 : to see the num of letter in word is plenty or not, Not = False
+    for letter, freq in check_word.items():
+        if letter not in letter_freq or freq > letter_freq[letter]:
+            return False
+
+    return True
 
 def score_word(word):
-    pass
+    if len(word) == 0 :
+        return 0
+
+    score = 0
+    # hardcode for score chart from README
+    # use it to get letters values(=scores)
+    score_chart = {
+        "A" : 1,
+        "B" : 3,
+        "C" : 3,
+        "D" : 2,
+        "E" : 1,
+        "F" : 4,
+        "G" : 2,
+        "H" : 4,
+        "I" : 1,
+        "J" : 8,
+        "K" : 5,
+        "L" : 1,
+        "M" : 3,
+        "N" : 1,
+        "O" : 1,
+        "P" : 3,
+        "Q" : 10,
+        "R" : 1,
+        "S" : 1,
+        "T" : 1,
+        "U" : 1,
+        "V" : 4,
+        "W" : 4,
+        "X" : 8,
+        "Y" : 4,
+        "Z" : 10
+    }
+
+    for letter in word:
+        upper_case = letter.upper()
+        if upper_case in score_chart:
+            score += score_chart[upper_case]
```

Comment:
Similar to the frequency dictionaries above, we could absolutely use the `get()` function to remove the conditional here! 
```suggestion
        score += score_chart.get(upper_case,0)
```